### PR TITLE
update of xsd (adding new "compression"-elements)

### PR DESCRIPTION
**Pull request by ptahmose**
## Description

* update of metadata-schema (adding some new elements related to "compression")
* add a changelog

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

locally, unittests

## Checklist:

- [x] I followed the Contributing Guidelines.
- [x] I did a self-review.
- [ ] I commented my code, particularly in hard-to-understand areas.
- [ ] I updated the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd [hunk 1]
```diff
@@ -7427,18 +7427,59 @@
           </xs:documentation>
         </xs:annotation>
       </xs:element>
-      <xs:element minOccurs="0" name="OriginalCompressionMethod" type="xs:string">
+      <xs:element name="OriginalCompressionMethod" type="xs:string" minOccurs="0">
         <xs:annotation>
           <xs:documentation>
-            The original compression method used when the file was initially saved.
+            This is an informative string specifying a compression method which is used (at subblock-level) when
```
Comment (DaveyJonesBitPail) [inline CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7433]: How about starting with: "The compression method used (at the subblock-level) when writing the CZI-Document"? We already have information that this element is a string from the type attribute.

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd [hunk 2]
```diff
@@ -7427,18 +7427,59 @@
...
             The original encoding quality (10..100) used when the file was initially saved and has used a compression method.
+            Note that this element is mostly present on legacy CZI files. Please refer to the
+            OriginalCompressionParameters element when saving the CZI - the encoding quality can be part of it.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:element>
+      <xs:element name="OriginalCompressionParameters" type="xs:string" minOccurs="0">
+        <xs:annotation>
+          <xs:documentation>
+            This is an informative string specifying the compression parameters which were used (at subblock-level) the
```
Comment (DaveyJonesBitPail) [inline CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7457]: See comment above about alternative start. The equivalent here would be
> The compression parameters which were used (at the subblock-level) during the first time the CZI-document was written.

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd [hunk 3]
```diff
@@ -7427,18 +7427,59 @@
...
+            alone.
             Use members of the CompressionMethods type for string content, e.g. "Jpg" or "JpgXr".
           </xs:documentation>
         </xs:annotation>
       </xs:element>
-      <xs:element minOccurs="0" name="OriginalEncodingQuality" type="xs:integer">
+      <xs:element name="OriginalEncodingQuality" type="xs:integer" minOccurs="0">
         <xs:annotation>
           <xs:documentation>
             The original encoding quality (10..100) used when the file was initially saved and has used a compression method.
```
Comment (DaveyJonesBitPail) [inline CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7448]: How about:
> The original encoding quality (10..100) used when the file was initially saved with a compression method.

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd [hunk 4]
```diff
@@ -7427,18 +7427,59 @@
...
+            alone.
+            The format should be key-value pairs, while each pair is separated with ',' and keys and values are
+            separated with ':', e.g. 'Lossless: False, Quality: 77'.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:element>
+      <xs:element name="CurrentCompressionParameters" type="xs:string" minOccurs="0">
+        <xs:annotation>
+          <xs:documentation>
+            This is an informative string specifying the compression parameters which were used (at subblock-level) the
```
Comment (DaveyJonesBitPail) [inline CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7473]: See above comment about alternative start to element documentation

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd [hunk 5]
```diff
@@ -7427,18 +7427,59 @@
...
+            last time the CZI-document was written. Note that conceptually the compression parameters are
```
Comment (DaveyJonesBitPail) [inline CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7474]:
```suggestion
            last time the CZI-document was written. Note that conceptually, the compression parameters are
```

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd [hunk 6]
```diff
@@ -7427,18 +7427,59 @@
...
+            Note that this element is mostly present on legacy CZI files. Please refer to the
+            OriginalCompressionParameters element when saving the CZI - the encoding quality can be part of it.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:element>
+      <xs:element name="OriginalCompressionParameters" type="xs:string" minOccurs="0">
+        <xs:annotation>
+          <xs:documentation>
+            This is an informative string specifying the compression parameters which were used (at subblock-level) the
+            first time the CZI-document was written. Note that conceptually the compression parameters are
```
Comment (DaveyJonesBitPail) [inline CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7458]:
```suggestion
            first time the CZI-document was written. Note that conceptually, the compression parameters are
```

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd [hunk 7]
```diff
@@ -7427,18 +7427,59 @@
           </xs:documentation>
         </xs:annotation>
       </xs:element>
-      <xs:element minOccurs="0" name="OriginalCompressionMethod" type="xs:string">
+      <xs:element name="OriginalCompressionMethod" type="xs:string" minOccurs="0">
         <xs:annotation>
           <xs:documentation>
-            The original compression method used when the file was initially saved.
+            This is an informative string specifying a compression method which is used (at subblock-level) when
+            writing the CZI-document. Note that conceptually the compression method (and it parametrization) is
```
Comment (DaveyJonesBitPail) [inline CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7434]:
```suggestion
            writing the CZI-document. Note that conceptually, the compression method (and it parametrization) is
```

File: CMakeLists.txt [hunk 8]
```diff
@@ -5,7 +5,7 @@
 cmake_minimum_required(VERSION 3.15)
 
 project(CZICheck 
-      VERSION 0.1.0
+      VERSION 0.1.1
```
Comment (DaveyJonesBitPail) [inline CMakeLists.txt:8]: Does a schema change not constitute a minor version update especially since a new element has been added?

**Review by DaveyJonesBitPail — APPROVED** [review]
Approved with suggestions/recommendations

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd (hunk 1 above)
Comment (ptahmose) [reply CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd:7433]: I am going to change "string" to "text". True, we know from the type attribute that this is a string, but I'd see a value in expressing the idea "the text does not have a defined format, it is free-form text, not intended to be parsed in any way". To express this "free form feature", we used the term "informative text/string" so far.

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd (hunk 2 above)
Comment (ptahmose) [reply CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd]: change to "text"

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd (hunk 5 above)
Comment (ptahmose) [reply CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd]: done

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd (hunk 6 above)
Comment (ptahmose) [reply CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd]: done

File: CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd (hunk 3 above)
Comment (ptahmose) [reply CZICheck/checkers/schema/ZEN/flatten/ImageMetadata.xsd]: done

**Review by DaveyJonesBitPail — APPROVED** [review]

**Pull request merged**